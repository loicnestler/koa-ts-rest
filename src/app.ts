import config from './config'

import Koa, { Context } from 'koa'
import KoaRespond from 'koa-respond'
import KoaLogger from 'koa-logger'
import KoaPinoLogger from 'koa-pino-logger'

import mongo from '@/database/mongo'

import { configureRoutes } from 'koa-joi-controllers'

import CookieHandler from '@/middlewares/cookiehandler'
import ErrorHandler from '@/middlewares/errorhandler'
import ValidationErrorHandler from '@/middlewares/validation-errorhandler'

import UserController from '@/controllers/user.controller'

// console.log(config.ENV)
const app: Koa = new Koa()
// app.use(ValidationErrorHandler())
app.use(config.ENV === 'development' ? KoaLogger() : config.ENV === 'test' ? async (ctx: Context, next: Function) => await next() : KoaPinoLogger())
app.use(KoaRespond())

app.use(CookieHandler())
app.use(ErrorHandler())
app.use(ValidationErrorHandler())


// TODO: add ratelimit middleware

configureRoutes(app, [
    new UserController,
])

const thread = mongo.connect(config.MONGO_URI).then(() => {
    console.log(`âœ… Successfully connected to MongoDB`)
    app.listen(config.PORT, () => {
        console.log(`ðŸš€ Listening on :${config.PORT}...`)
    })
}).catch(err => {
    console.log(err)
})

export { thread, app }