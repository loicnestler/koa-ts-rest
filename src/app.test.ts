import request from 'supertest'
import { thread, app } from './app'


// beforeAll(async () => {
//     await thread
// })

describe('Server', () => {
    it('should start', async () => {
        const response = await request(app.callback()).get('/')
        expect(response.status).toBeGreaterThan(1)
    })

    it('should throw 404 on /', async () => {
        const response = await request(app.callback()).get('/')
        expect(response.status).toBe(404)
        expect(response.text).toMatchSnapshot()

    })
})

// test('Server', async () => {
//     describe('should start', () => {

//     })
// })