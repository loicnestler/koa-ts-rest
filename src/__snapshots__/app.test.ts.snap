// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server should start 1`] = `"{\\"error\\":true,\\"message\\":\\"not found\\"}"`;

exports[`Server should throw 404 on / 1`] = `"{\\"error\\":true,\\"message\\":\\"not found\\"}"`;
